### PR TITLE
Add ID field if it's missing from query spec

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

**Comment by bhariharan** [comment]
@wmathurin I'm considering doing something similar for ```LAST_MODIFIED_DATE``` if it's missing, since that's special too. What do you think?

**Comment by wmathurin** [comment]
Should we add a test?

File: libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SoqlSyncDownTarget.java [hunk 1]
```diff
@@ -65,8 +68,20 @@ public SoqlSyncDownTarget(String query) {
...
 	}
-	
+
+    private void addIdFieldIfRequired() {
+        if (!TextUtils.isEmpty(query)) {
+            final String idFieldName = getIdFieldName();
+
+            // Inserts the mandatory 'Id' field if it doesn't exist.
+            if (!query.contains(idFieldName)) {
+                query = query.replaceFirst("([sS][eE][lL][eE][cC][tT] )", "select " + idFieldName + ", ");
```
Comment (wmathurin) [inline libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SoqlSyncDownTarget.java]: I guess it should work even with subqueries (https://developer.salesforce.com/docs/atlas.en-us.200.0.soql_sosl.meta/soql_sosl/sforce_api_calls_soql_select.htm) thanks to the replaceFirst.

Comment (bhariharan) [inline libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SoqlSyncDownTarget.java]: Yes it should.

**Comment by bhariharan** [comment]
Sure - I'll add a test. The way it's setup now, ```SmartSyncExplorer``` will fail if this stops working, because I removed ```Id``` from its index specs.

**Comment by bhariharan** [comment]
Adding ```LAST_MODIFIED_DATE``` [here](https://github.com/bhariharan/SalesforceMobileSDK-Android/commit/67982df2a475fe742bc4620ff73d7010d632724a).

**Comment by wmathurin** [comment]
:+1:

**Comment by bhariharan** [comment]
Added test [here](https://github.com/bhariharan/SalesforceMobileSDK-Android/commit/8c8f86387d0ee486dd9cd4f2f294b804af118fa5).

**Pull request merged**